omod/src/main/java/org/openmrs/module/aijar/web/controller/AppointmentCountController.java

```diff
@@ -0,0 +1,63 @@
+
+package org.openmrs.module.aijar.web.controller;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+public class AppointmentCountController extends BaseRestController {
+
+    AdministrationService administrationService = Context.getAdministrationService();
+
+    //@RequestMapping(method = RequestMethod.GET)
+    @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/ugandaemr/appointmentcount",method = RequestMethod.GET)
+    @ResponseBody
+    public SimpleObject getNumberOfPatients(
+            @RequestParam(required = true, value = "nextAppointmentDate") Date nextAppointmentDate,
+            @RequestParam(value = "encounterType", defaultValue = "") String encounterType) {
+
+        SimpleObject response = new SimpleObject();
+
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+        String returnDate = formatter.format(nextAppointmentDate);
+        List patients = new ArrayList();
+
+        if (StringUtils.isEmpty(encounterType)) {
+            String sql = "SELECT obs.person_id FROM obs WHERE obs.concept_id=5096 AND obs.value_datetime='" + returnDate
+                    + "' AND obs.voided = 0";
+            patients = administrationService.executeSQL(sql, true);
+
+        } else {
+            String[] encounterT = encounterType.split(",");
```

Comment:
@reagan-meant This should create a comma delimited list of encounterTypes such as ('encounterTypeuuid1', 'encounterTypeuuid2') so that you only run one query

Reply:
@reagan-meant here is potential code that needs to be tested 

`String sql = "SELECT obs.person_id FROM obs WHERE obs.concept_id=5096 AND obs.value_datetime='" + returnDate
                + "' AND obs.voided = 0";
        // the where clause filter for encounter types - defaults to an empty string
        String encounterFilterWhereClause = "";
        if (!StringUtils.isEmpty(encounterType)) {
            encounterFilterWhereClause = "  AND obs.encounter_id IN ('" + StringUtils.join(encounterType.split(","), "','") + "')";
        }
        patients = administrationService.executeSQL(sql + encounterFilterWhereClause, true);`